merchant/src/catalog/backend.rs

```diff
@@ -158,6 +219,98 @@ impl CatalogService for CatalogSQLService {
+                }
+                _ => {}
+            }
+        }
+
+        let mut items_sorted_to_insert: Vec<(&String, &u32)> = objects_dependency.iter().collect();
+        items_sorted_to_insert.sort_by(|a, b| b.1.cmp(a.1));
+        // we start creating the dependencies from the less dependant
+        for (alias_id, _) in items_sorted_to_insert.clone() {
+            let item = *objects_dependency_document.get(alias_id).unwrap();
+            match item {
+                CatalogObject::Item(item) => {
+                    let catalog_object: CatalogObject<self::Id> =
+                        CatalogObject::<u32>::Item(item.clone());
+                    let document = self.create(account.clone(), &catalog_object).await?;
+                    objects_created.entry(alias_id.clone()).or_insert(document);
+                }
+                CatalogObject::Variation(variation) => {
+                    if let Some(catalog_object) = objects_created.get(&variation.item_id.clone()) {
+                        let catalog_object =
+                            <Self as BulkDocumentConverter>::to_simple_catalog_object(
+                                catalog_object.id,
+                                &CatalogObject::Variation(variation.clone()),
+                            )
+                            .unwrap();
+                        let document = self.create(account.clone(), &catalog_object).await?;
+                        objects_created.entry(alias_id.clone()).or_insert(document);
+                    } else {
+                        return Err(CatalogError::BulkReferenceNotExist(
+                            variation.item_id.clone(),
+                        ));
+                    }
+                }
+                CatalogObject::Modification(modification) => {
+                    if let Some(catalog_object  ) = objects_created.get(&modification.item_id.clone())
+                    {
+                        let catalog_object =
+                            <Self as BulkDocumentConverter>::to_simple_catalog_object(
+                                catalog_object.id,
+                                &CatalogObject::Modification(modification.clone()),
+                            )
+                            .unwrap();
+                        let document = self.create(account.clone(), &catalog_object).await?;
+                        objects_created.entry(alias_id.clone()).or_insert(document);
+                    } else {
+                        return Err(CatalogError::BulkReferenceNotExist(
+                            modification.item_id.clone(),
+                        ));
+                    }
+                }
+            }
+        }
+
+        let objects_created: Vec<SqlCatalogObjectDocument> = items_sorted_to_insert
+            .iter()
+            .map(|(id, _)| objects_created.remove(&(*id).clone()).unwrap())
+            .collect();
```

Comment:
One last cosmetic touch, I also think you can `collect` directly into the result of the return type